core/src/main/java/bisq/core/btc/model/BsqTransferModel.java

```diff
@@ -0,0 +1,77 @@
+package bisq.core.btc.model;
+
+import bisq.core.dao.state.model.blockchain.TxType;
+import bisq.core.util.coin.CoinUtil;
+
+import org.bitcoinj.core.Coin;
+import org.bitcoinj.core.LegacyAddress;
+import org.bitcoinj.core.Transaction;
+
+import lombok.Getter;
+
+@Getter
+public final class BsqTransferModel {
+
+    private final LegacyAddress receiverAddress;
+    private final Coin receiverAmount;
+    private final Transaction preparedSendTx;
+    private final Transaction txWithBtcFee;
+    private final Transaction signedTx;
+    private final Coin miningFee;
+    private final int txSize;
+    private final TxType txType;
+
+    public BsqTransferModel(LegacyAddress receiverAddress,
+                            Coin receiverAmount,
+                            Transaction preparedSendTx,
+                            Transaction txWithBtcFee,
+                            Transaction signedTx) {
+        this.receiverAddress = receiverAddress;
+        this.receiverAmount = receiverAmount;
+        this.preparedSendTx = preparedSendTx;
+        this.txWithBtcFee = txWithBtcFee;
+        this.signedTx = signedTx;
+        this.miningFee = signedTx.getFee();
+        this.txSize = signedTx.bitcoinSerialize().length;
+        this.txType = TxType.TRANSFER_BSQ;
+    }
+
+    public String getReceiverAddressAsString() {
+        return receiverAddress.toString();
+    }
+
+    public double getMiningFeeInSatoshisPerByte() {
```

Comment:
After much discussion on keybase, changed amount type to String in commit 65df9e1503a9a38518ac29544e8a62cd4ff1045a.

A user needs to be able to enter a representation of a decimal value like "0.25" instead of a satoshi value like 25000000.
The server will convert the string to a Coin for use in a transaction.

Reply:
The CoinUtil method is uses returns a double, but deleted this almost unused method anyway in commit 2efd094f68f2b867624a7c27191b42a08bb15f8f.